cachito/workers/pkg_managers/rubygems.py

```diff
@@ -400,10 +405,34 @@ def unpack_git_dependency(dep):
     """
     #
     extracted_path = str(dep["path"]).removesuffix(".tar.gz")
+    log.debug(f"Extracting archive at {dep['path']} to {extracted_path}")
     shutil.unpack_archive(dep["path"], extracted_path)
     os.remove(dep["path"])
     dep["path"] = extracted_path
 
+    if dep["branch"] != "":
+        log.debug(f"Checking out branch {dep['branch']} at {dep['path']}/app")
```

Comment:
For concatenating file paths, it's usually better to use the the library rather than doing it with strings
```suggestion
        log.debug(f"Checking out branch {dep['branch']} at {dep['path'] / 'app'}")
```
(`dep['path']` is a Path, right?)

Doesn't matter as much for the log message, but it's done this way in other places in the PR as well